src/lang/builtins_thread.ml

```diff
@@ -0,0 +1,87 @@
+(*****************************************************************************
+
+  Liquidsoap, a programmable audio stream generator.
+  Copyright 2003-2019 Savonet team
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details, fully stated in the COPYING
+  file at the root of the liquidsoap distribution.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301  USA
+
+ *****************************************************************************)
+
+open Lang_builtins
+
+let () =
+  add_builtin "thread.run" ~cat:Control
+    [
+      "fast", Lang.bool_t, Some (Lang.bool true),
+      Some "Whether the thread is supposed to quickly or not. Typically, \
+            blocking tasks (e.g. fetching data over the internet) should not be \
+            considered to be fast. When set to `false` its priority will be \
+            lowered below that of request resolutions and fast timeouts. This is \
+            only effective if you set a dedicated queue for fast tasks, see the \
+            \"scheduler\" settings for more details." ;
+      "delay", Lang.float_t, Some (Lang.float 0.),
+      Some "Delay (in sec.) after which the thread should be lauched.";
+      "", Lang.fun_t [] Lang.unit_t, None, Some "Function to execute."
+    ]
+    Lang.unit_t
+    ~descr:"Run a function in a separate thread."
```

Comment:
This is not true. `duppy` runs a pool of threads who pick and execute tasks similarly to the way `node` is implemented. There isn't a single thread per tasks, which was the case before and the reason we introduced duppy. I believe that using `tasks` (or `task`) instead of `thread` would be more appropriate.

Reply:
I do agree with you, but to me this is only an implementation detail: the average user wants to know that there are something like threads in Liq, and "task" is not very clear nor common...